### PR TITLE
Alerts of CRUD

### PR DESCRIPTION
**Pull request by gayanDeegala**
closing #71

### CODE REVIEW

**Comment by malindakpt** [comment]
There are still some default alerts for cases like delete properties. Please replace them also.

File: src/core/CogniteJsonEditorOptions.ts [hunk 1]
```diff
@@ -130,19 +131,19 @@ export class CogniteJsonEditorOptions implements JSONEditorOptions {
                             case ErrorType.RequiredNode:
                                 item.title = "Cannot Remove. This field is mandatory";
                                 item.click = () => {
-                                    alert("Error: Cannot Remove. This field is mandatory");
+                                    message.error("Cannot Remove. This field is mandatory");
```
Comment (mihilranathunga) [inline src/core/CogniteJsonEditorOptions.ts:134]: Move these messages to a constant file

File: src/core/CogniteJsonEditorOptions.ts [hunk 2]
```diff
@@ -130,19 +131,19 @@ export class CogniteJsonEditorOptions implements JSONEditorOptions {
...
                                 }
                                 break;
                             case ErrorType.MinLength:
                                 item.title = "Cannot Remove. Array has a minimum length";
                                 item.click = () => {
-                                    alert("Error: Cannot Remove. Array has a minimum length");
+                                    message.error("Cannot Remove. Array has a minimum length");
```
Comment (mihilranathunga) [inline src/core/CogniteJsonEditorOptions.ts:140]: same here

File: src/core/CogniteJsonEditorOptions.ts [hunk 3]
```diff
@@ -130,19 +131,19 @@ export class CogniteJsonEditorOptions implements JSONEditorOptions {
...
                                 }
                                 break;
                             default:
                                 item.title = "Cannot Remove.";
                                 item.click = () => {
-                                    alert("Error: Cannot Remove.");
+                                    message.error("Cannot Remove.");
```
Comment (mihilranathunga) [inline src/core/CogniteJsonEditorOptions.ts:146]: same here

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 4]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
-                    JsonConfigCommandCenter.errorAlert("Save Cancelled!", error);
-                });
+        if (!isValidFileName()) {
+            message.error("Save Cancelled!\nPlease add a file name");
+        }
+        else {
+            confirm({
+                title: 'Cretate New Json Config',
+                icon: <ExclamationCircleOutlined/>,
+                content: 'Do you want to cretate new Json Config?',
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:43]: every hard coded string should be in constants

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 5]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
-                .then((response: any) => {
-                    const createdId = response.data.data.items[0].id;
-                    props.reloadJsonConfigs(createdId);
-                    alert("Data saved successfully!");
-                })
-                .catch((error: any) => {
-                    JsonConfigCommandCenter.errorAlert("Save Cancelled!", error);
-                });
+        if (!isValidFileName()) {
+            message.error("Save Cancelled!\nPlease add a file name");
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:37]: same here

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 6]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
+            confirm({
+                title: 'Cretate New Json Config',
+                icon: <ExclamationCircleOutlined/>,
+                content: 'Do you want to cretate new Json Config?',
+                onOk() {
+                    props.commandEvent(CommandEvent.saveAs)
+                        .then((response: any) => {
+                            const createdId = response.data.data.items[0].id;
+                            props.reloadJsonConfigs(createdId);
+                            message.success("Data saved successfully!");
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:49]: same here

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 7]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
+                        })
+                        .catch((error: any) => {
+                            message.error(`Save Cancelled! ${extractErrorMessage(error)}`);
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:52]: same here

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 8]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
-                    JsonConfigCommandCenter.errorAlert("Update failed!", error);
-                });
+        if (!isValidFileName()) {
+            message.error("Update Cancelled!\nPlease select a file");
+        }
+        else {
+            confirm({
+                title: 'Update File',
+                icon: <ExclamationCircleOutlined/>,
+                content: 'Do you want to update file with new changes?',
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:67]: same here

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 9]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
-                })
-                .catch((error: any) => {
-                    JsonConfigCommandCenter.errorAlert("Update failed!", error);
-                });
+        if (!isValidFileName()) {
+            message.error("Update Cancelled!\nPlease select a file");
+        }
+        else {
+            confirm({
+                title: 'Update File',
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:65]: move this to constants

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 10]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
+                content: 'Do you want to update file with new changes?',
+                onOk() {
+                    props.commandEvent(CommandEvent.update)
+                        .then((response: any) => {
+                            const createdId = response.data.data.items[0].id;
+                            props.reloadJsonConfigs(createdId);
+                            message.success("Data updated successfully!");
+                        })
+                        .catch((error: any) => {
+                            message.error(`Update failed! ${extractErrorMessage(error)}`);
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:76]: move hardoced strings to constants

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 11]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
-            props.commandEvent(CommandEvent.delete)
-                .then(() => {
-                    props.reloadJsonConfigs(null);
-                    alert("Json Config Deleted successfully!");
-                })
-                .catch((error: any) => {
-                    JsonConfigCommandCenter.errorAlert("Delete Cancelled!", error);
-                });
+        if (!isValidFileName()) {
+            message.error("Delete Cancelled!\nPlease select a file");
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:85]: move strings to constants

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 12]
```diff
@@ -20,43 +33,73 @@ export const CommandPanel: React.FC<{
...
+            confirm({
+                title: 'Delete Config',
+                icon: <ExclamationCircleOutlined/>,
+                content: 'Are you sure you want to permanently delete this config?',
+                onOk() {
+                    props.commandEvent(CommandEvent.delete)
+                        .then(() => {
+                            props.reloadJsonConfigs(null);
+                            message.success("Json Config Deleted successfully!");
+                        })
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:97]: same here

File: src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx [hunk 13]
```diff
@@ -18,8 +25,7 @@ export const JsonConfigurator: React.FC<any> = () => {
                 setJsonConfigMap(response);
             })
             .catch(error => {
-                console.error("Retrieved Digital Twin List failed!");
-                JsonConfigCommandCenter.errorAlert("Save Cancelled!", error);
+                message.error(`Unable to retrieve json configs! ${extractErrorMessage(error)}`);
```
Comment (mihilranathunga) [inline src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx:28]: move hard coded strings to constants

File: src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx [hunk 14]
```diff
@@ -6,6 +6,13 @@ import { JsonConfigCommandCenter } from "../../../core/JsonConfigCommandCenter";
 import { CommandPanel } from "../CommandPanel/CommandPanel";
 import { SideNavPanel } from '../SideNavPanel/SideNavPanel';
 import { EditorPanel } from '../EditorPanel/EditorPanel';
+import { message } from 'antd';
```
Comment (mihilranathunga) [inline src/userInterface/panels/JsonConfigurator/JsonConfigurator.tsx:9]: use specific import whenere possible to reduce library weight like "antd/es/message"

**Review by mihilranathunga — CHANGES_REQUESTED** [review]
move strings to constants

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 15]
```diff
@@ -24,43 +38,73 @@ export const CommandPanel: React.FC<{
...
+            confirm({
+                title: `${LOCALIZATION.UPLOAD_TITLE}`,
+                icon: <ExclamationCircleOutlined />,
+                content: `${LOCALIZATION.UPLOAD_CONTENT}`,
+                onOk() {
+                    props.commandEvent(CommandEvent.update)
+                        .then((response: any) => {
+                            const createdId = response.data.data.items[0].id;
+                            props.reloadJsonConfigs(createdId);
+                            message.success(`${LOCALIZATION.UPLOAD_SUCCESS}`);
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:78]: can't you put this as message.success(LOCALIZATION.UPLOAD_SUCCESS);

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx [hunk 16]
```diff
@@ -1,10 +1,24 @@
 import React from 'react';
 import classes from './CommandPanel.module.scss';
 import { CommandItem } from '../../components/CommandItem/CommandItem';
-import { Switch } from "antd";
+import { Switch } from 'antd/es';
```
Comment (mihilranathunga) [inline src/userInterface/panels/CommandPanel/CommandPanel.tsx:4]: these should be antd/es/switch

**Review by mihilranathunga — APPROVED** [review]

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx (hunk 16 above)
Comment (gayanDeegala) [reply src/userInterface/panels/CommandPanel/CommandPanel.tsx:4]: 'antd/es/switch' has no exported member 'Switch'
with 'antd/es/switch' we can't use named imports

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx (hunk 15 above)
Comment (gayanDeegala) [reply src/userInterface/panels/CommandPanel/CommandPanel.tsx]: yup. fixed it

File: src/userInterface/panels/CommandPanel/CommandPanel.tsx (hunk 16 above)
Comment (gayanDeegala) [reply src/userInterface/panels/CommandPanel/CommandPanel.tsx]: changing to default imports

**Review by mihilranathunga — APPROVED** [review]

**Pull request merged**